### PR TITLE
Removing the optimization that converts "power(X, 0.5)" to "sqrt(X)", plus fix rocBLAS hgemm API call site

### PR DESCRIPTION
**Pull request by deven-amd**

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/elemental_ir_emitter.cc [hunk 1]
```diff
@@ -198,36 +198,6 @@ StatusOr<llvm::Value*> GpuElementalIrEmitter::EmitPowerOp(
   PrimitiveType output_type = op->shape().element_type();
   llvm::Type* llvm_ty = lhs_value->getType();
 
-  auto make_sqrt = [&, this]() -> StatusOr<llvm::Value*> {
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/elemental_ir_emitter.cc]: this is unlikely not upstreamble. how should we fix it in future PRs?

Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/elemental_ir_emitter.cc]: oh, so the code is gone in upstream.

**Comment by deven-amd** [comment]
jenkins : retest this please

**Comment by whchung** [comment]
@deven-amd before we have more in-depth knowledge about the two failing cases:
```
[  FAILED  ] DotOperationTest_F16F32F64CF64/0.GeneralMatMulR3LhsR2Rhs, where TypeParam = Eigen::half                                                                                                              
[  FAILED  ] DotOperationTest_F16F32F64CF64/0.GeneralMatMulR2LhsR3Rhs, where TypeParam = Eigen::half                                                                                                              
```
should we mark the as `DISABLED_ON_GPU_ROCM`? These cases don't seem to be related to the changes introduced in this PR.

**Comment by whchung** [comment]
Jenkins: retest this please

**Comment by whchung** [comment]
`rocm-xla` is passing. merging this PR.

**Pull request merged**